rasa/graph_components/validators/config_validator.py

```diff
@@ -0,0 +1,532 @@
+        self._warn_if_priorities_are_not_unique()
+        self._warn_if_rule_based_data_is_unused_or_missing(story_graph=story_graph)
+
+    def _warn_if_no_rule_policy_is_contained(self) -> None:
+        """Warns if there is no rule policy among the given policies.
+
+        NOTE: core/policies/_ensemble/check_for_important_policies
+        """
+        if not any(
+            node.uses == RulePolicyGraphComponent for node in self._policy_schema_nodes
+        ):
+            rasa.shared.utils.io.raise_warning(
+                f"'{RulePolicyGraphComponent.__name__}' is not included in the model's "
+                f"policy configuration. Default intents such as "
+                f"'{USER_INTENT_RESTART}' and '{USER_INTENT_BACK}' will "
+                f"not trigger actions '{ACTION_RESTART_NAME}' and "
+                f"'{ACTION_BACK_NAME}'.",
+                docs=DOCS_URL_DEFAULT_ACTIONS,
+            )
+
+    def _raise_if_domain_contains_form_names_but_no_rule_policy_given(
+        self, domain: Domain,
+    ) -> None:
+        """Validates that there exists a rule policy if forms are defined.
+
+        NOTE: _ensemble/check_domain_ensemble_compatibility (called from agent.py)
+
+        Raises:
+            `InvalidConfigException` if domain and rule policies do not match
+        """
+        contains_rule_policy = any(
+            schema_node
+            for schema_node in self._graph_schema.nodes.values()
+            if schema_node.uses == RulePolicyGraphComponent
+        )
+
+        if domain.form_names and not contains_rule_policy:
+            raise InvalidDomain(
+                "You have defined a form action, but have not added the "
+                f"'{RulePolicyGraphComponent.__name__}' to your policy ensemble. "
+                f"Either remove all forms from your domain or add the "
+                f"'{RulePolicyGraphComponent.__name__}' to your policy configuration."
+            )
+
+    def _raise_if_a_rule_policy_is_incompatible_with_domain(
+        self, domain: Domain,
+    ) -> None:
+        """Validates the rule policies against the domain.
+
+        NOTE: _ensemble/check_domain_ensemble_compatibility (called from agent.py)
+
+        Raises:
+            `InvalidDomain` if domain and rule policies do not match
+        """
+        for schema_node in self._graph_schema.nodes.values():
+            if schema_node.uses == RulePolicyGraphComponent:
+                RulePolicyGraphComponent.raise_if_incompatible_with_domain(
+                    config=schema_node.config, domain=domain
+                )
+
+    def _warn_if_priorities_are_not_unique(self) -> None:
+        """Warns if the priorities of the policies are not unique.
+
+        NOTE: _ensemble/check_priorities
+        """
+        priority_dict = defaultdict(list)
+        for schema_node in self._policy_schema_nodes:
+            if "priority" in schema_node.config:
```

Comment:
don't we need to get the default priority for the policy otherwise?

Reply:
ouch, yep